### PR TITLE
extract some React components

### PR DESCRIPTION
**Pull request by bostonaholic**
this helps in readability of the `ContainerHomePreview` component.

### CODE REVIEW

**Comment by bostonaholic** [comment]
This is a precursor to me working on #1455

**Comment by FrenchBen** [comment]
Looks good so far, but it currently breaks the `click` functionality.
Aka you cannot click on the preview, the launch external icon nor the gear icon to see the settings.

Can you fix the above and we can merge this? I like the separation of the components 👍

**Comment by bostonaholic** [comment]
Good catch @FrenchBen. Fixed.

**Comment by FrenchBen** [comment]
Can you squash the last 3 commits, since they're pretty much all to fix 1 thing?

File: src/components/ContainerHomePreview.react.js [hunk 1]
```diff
@@ -29,15 +29,6 @@ var ContainerHomePreview = React.createClass({
...
-  handleClickPreview: function () {
-    if (this.props.defaultPort) {
-      metrics.track('Opened In Browser', {
-        from: 'preview'
-      });
-      shell.openExternal('http://' + this.props.ports[this.props.defaultPort].url);
-    }
-  },
-
   handleClickNotShowingCorrectly: function () {
```
Comment (FrenchBen) [inline src/components/ContainerHomePreview.react.js]: Since we're re-writing this piece, can this be renamed to something like `handleClickPortSettings`?

Comment (bostonaholic) [inline src/components/ContainerHomePreview.react.js]: @FrenchBen I'm so glad you suggested that. I was reluctant to rename it myself as much as I wanted to.

Fixed and squashed.

**Pull request merged**